infra/firebase_client_mock.go

```diff
@@ -0,0 +1,25 @@
+package infra
+
+import (
+	"context"
+
+	"firebase.google.com/go/v4/auth"
+)
+
+type mockedTokenVerifier struct{}
+
+func (m mockedTokenVerifier) VerifyIDToken(_ context.Context, email string) (*auth.Token, error) {
```

Comment:
Simple stubbed firebase api client (in fact: token verifier interface, because that is what we need) for usage in tests. I simplified things a bit by making it take an email string as input, rather than rewriting a lot of "parse a jwt token" logic.